### PR TITLE
feat(Menu): MenuItemLink component

### PR DESCRIPTION
**Pull request by miroslavstastny**
This is a Proof-of-Concept of splitting `MenuItem` component which renders as `li > a`  to `MenuItem` which renders as `li` and `MenuItemLink` which renders as `a`.
The main motivation for this are issues #79 and #191.

**BEWARE: This is just POC with dirty code and disabled UT's, it must be reworked if we decide to take this path!**

All existing Menu examples render without any change with no visual regressions.

## API

Consider following Menu example which uses MenuItem shorthands:
```jsx
<Menu defaultActiveIndex={0} items={items} />
```

`MenuItem.content` is a shorthand for `MenuItemLink`.

:+1: It can be a string `{ content: 'Home' }`:
```jsx
<li class="ui-menu__item" role="presentation">
  <a class="ui-menu__itemlink" role="menuitem" tabindex="0" data-is-focusable="true">Home</a>
</li>
```

:+1: It can be an object `{ content: {content: 'Home'}}` - the same result as the previous one:
```jsx
<li class="ui-menu__item" role="presentation">
  <a class="ui-menu__itemlink" role="menuitem" tabindex="0" data-is-focusable="true">Home</a>
</li>
```

:+1: It can contain icon (shorthand) `{ content: {content: 'Home', icon: 'home'}}`:
```jsx
<li class="ui-menu__item" role="presentation">
  <a class="ui-menu__itemlink" role="menuitem" tabindex="0" data-is-focusable="true">
    <span class="ui-icon" aria-hidden="true"></span>
    Home
  </a>
</li>
```

:+1: It can contain icon only `{ content: {icon: 'home' }}`:
```jsx
<li class="ui-menu__item" role="presentation">
  <a class="ui-menu__itemlink" role="menuitem" tabindex="0" data-is-focusable="true">
    <span class="ui-icon" aria-hidden="true"></span>
  </a>
</li>
```

:-1: For backwards compatibility, the icon can be specified directly as MenuItem prop `{ content: {content: 'Home'}, icon: 'home' }`.
This can be confusing as icon can be specified both in content and aside the content (the one in content takes precedence). This functionality can be removed if desired.

:+1: It is possible to specify href (and other link attributes for the `a` element `{ content: {content: 'github.com', href:'https://github.com'}}`:
```jsx
<li class="ui-menu__item" role="presentation">
  <a class="ui-menu__itemlink" role="menuitem" tabindex="0" data-is-focusable="true" href="https://github.com">github.com</a>
</li>
```

:+1: It is possible to specify `as` for the inner element `{ content: {as: Button, content:'Home', icon: 'home'}}`:
```jsx
<li class="ui-menu__item" role="presentation">
  <button class="ui-button ui-menu__itemlink" role="menuitem" aria-disabled="false" tabindex="0" data-is-focusable="true">
    <span class="ui-icon" aria-hidden="true"></span>
    Home
  </button>
</li>
```

:-1: It is possible to provide own element as content, but that breaks accessibility `{ content: <Button content="Home" icon="home" /> }`:
```jsx
<li class="ui-menu__item" role="presentation">
  <button class="ui-button" aria-disabled="false" index="10">
    <span class="ui-icon" aria-hidden="true"></span>
    <span>Home</span>
  </button>
</li>
```

### CODE REVIEW

**Review by levithomason — APPROVED** [review]
After tests, looks great!

File: src/components/Menu/MenuItemLink.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    'disabled',
+    'icon',
+    'iconOnly',
+    'index',
+    'onClick',
+    'pills',
+    'pointing',
+    'styles',
+    'type',
+    'underlined',
```
Comment (alinais) [inline src/components/Menu/MenuItemLink.tsx]: Per latest updates, we don't need to specify the handledProps anymore.

File: src/components/Menu/MenuItemLink.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  static create: Function
+
+  static propTypes = {
+    /** A menu item can be active. */
+    active: PropTypes.bool,
+
+    /** An element type to render as (string or function). */
+    as: customPropTypes.as,
+
+    /** Primary content. */
```
Comment (alinais) [inline src/components/Menu/MenuItemLink.tsx]: Please add `@docSiteIgnore` to the comment so that it won't show in the doc site approved props for users.

**Review by alinais — CHANGES_REQUESTED** [review]

**Comment by miroslavstastny** [comment]
Closing, deprecated by #323.

**Pull request closed without merging**